src/Visitors/VisitorCollector.php

```diff
@@ -154,7 +166,7 @@ private function wrapper(
         ) {
             $beforeProblems = count($problemsHolder->getProblems());
 
-            if ($this->shouldSkip($node, $ancestors, $inspection)) {
+            if ($this->shouldSkip($node, $parent, $ancestors, $inspection)) {
```

Comment:
The core of the problem was that the `$parent` was not considered here at all, thus not included in the path that is relevant for `IgnoreByNameSuppressorInspection`. The given `$ancestors` only contains all parents excluding `$parent`.